tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts

```diff
@@ -0,0 +1,196 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {scaleLinear, scaleLog} from 'd3';
+
+import {ScaleType} from './scale_types';
+
+export interface Scale {
+  forward(domain: [number, number], range: [number, number], x: number): number;
+
+  reverse(domain: [number, number], range: [number, number], x: number): number;
+
+  nice(minAndMax: [number, number]): [number, number];
+
+  /**
+   * @param lowAndHigh bounds of the tick
```

Comment:
Unclear: what are these? Values in the domain? Values in the range?
Bounds of a single tick (like, the thing that’s 1–2 px wide)? Is this
just the extent of the axis, or only counting from the first tick mark?
What about the results—are they in domain-space or range-space?

This method seems useful and helpful; just a bit more clarity here would
be nice.

Reply:
Also, `minAndMax` but `lowAndHigh`—do they mean the same thing? If so,
let’s be consistent. (And descriptive: perhaps `domain`?)

Reply:
Changed everything to `domain`.